dias/analyzers/autos_analyzer.py

```diff
@@ -0,0 +1,393 @@
+            f.create_dataset('rms_noise', data=rms_noise, dtype=float)
+            f.create_dataset('axis/freq', data=freq, dtype=float)
+            f.create_dataset('axis/input', data=np.arange(NINPUT), dtype=int)
+            f.close()
+
+        self.run_timer.set(int(tt.time() - run_start_time))
+
+    def fit_autos(self, start_time, end_time):
+        """Load autocorrelations from the most recent transit.
+
+        Loop over frequencies and inputs and get fit paramters.
+
+        Parameters
+        ----------
+        start_time : datetime.datetime
+            Search for transits datetime (in UTC).
+        end_time : datetime.datetime
+            Search for transits before this datetime (in UTC).
+
+        Returns
+        -------
+        centroid_wander : np.ndarray
+            2D array of type `float` with axes `[freq, input]` containing the
+            difference of the transit centroid from the expected transit time in seconds.
+        width : np.ndarray
+            2D array of type `float` with axes `[freq, input]` containing the FWHM in seconds.
+        gain : np.ndarray
+            2D array of type `float` with axes `[freq, input]` containing the ADC-to-JY calibration
+            factor calculated by dividing the expected source flux by the fitted Gaussian peak.
+        sefd : np.ndarray
+            2D array of type `float` with axes `[freq, input]` containing the SEFD in Jy during the
+            transit. Calculated by fitting and calibrating the background around transit.
+        rms_noise : np.ndarray
+            2D array of type `float` with axes `[freq, input]` containing the root mean
+            square noise in Jy.
+        freq : list
+            Centre frequencies in MHz.
+        """
+        # Set a Finder object
+        f = self.Finder()
+        f.set_time_range(start_time, end_time)
+
+        # get autos during for particular transit
+        f.filter_acqs((data_index.ArchiveInst.name == 'chimecal'))
+        f.accept_all_global_flags()
+        f.include_transits(self.sel_source, time_delta=800.)
+
+        results_list = f.get_results()
+
+        if not results_list:
+            self.logger.warn(
+                    'Did not find any data in the archive for source ' + self.source)
+            return (None, None)
+        else:
+            result = results_list[0]
+            self.logger.info(
+                'Found ' + str(len(results_list)) + ' transits.')
+        self.transits_found.set(len(results_list))
+
+        result = results_list[-1]
+
+        nfreq = len(self.freq_sel)
+        self.logger.info(
+                'Loading autocorrelations at ' + str(nfreq) + ' frequencies.')
+        data = andata.CorrData.from_acq_h5(result[0], freq_sel=self.freq_sel)
+
+        times = data.index_map['time']['ctime']
+        freq = data.index_map['freq']['centre']
+        input_list = data.index_map['input']['chan_id']
+        self.freq_width = data.index_map['freq']['width']
```

Comment:
_*_ `freq_width` is only used in `fit_params`, so it is sufficient to pass it with the function call instead of making it a member of the class.

A class member stays in memory until the object is destroyed, so would be kept between task runs in this case.